docs_src/microservices/general/messagebus.md

```diff
@@ -0,0 +1,368 @@
+# EdgeX MessageBus
+
+## Introduction
+
+Edgex has an internal message bus referred to as the **EdgeX MessageBus** , which is used for internal communications between EdgeX services. An EdgeX Service is any Core/Support/Application/Device Service from EdgeX or any custom Application or Device Service built with the EdgeX SDKs. 
+
+The following diagram shows how each of the EdgeX Service use the EdgeX MessageBus.
+
+![[Insert Image of EdgeX MessageBus]](./messagebus diagram.jpg)
+
+The EdgeX MessageBus is meant for internal EdgeX service to service communications. It is not meant as an entry point for external services to communicate with the internal EdgeX services. The eKuiper Rules Engine is an exception to this as it is tightly integrated with EdgeX.
+
+There are EdgeX services which are meant as external entry points, which are:
+
+- **REST API on all the EdgeX services** - Accessed directly in non-secure mode or via the [API Gateway](../../../security/Ch-APIGateway) when running in secure mode
+
+- **App Service using External MQTT Trigger** - An App Service configured to use the [External MQTT Trigger](../../application/Triggers/#external-mqtt-trigger) will accept data from external services on an "external" MQTT connection
+  
+- **App Service using HTTP Trigger** - An App Service configured to use the [HTTP Trigger](../../application/Triggers/#http-trigger) will accept data from external services on an "external" REST connection. Accessed in the same manner as other EdgeX REST APIs.
+
+- **App Service using Custom Trigger** - An App Service configured to use a [Custom Trigger](../../application/Triggers/#custom-trigger) can accept data from external services or over additional protocols with few limitations.

```

Comment:
do we have some named example for this?

Reply:
Good idea there's an example with trigger that reads from stdout we could link.

Reply:
Yep, here: https://github.com/edgexfoundry/edgex-examples/tree/main/application-services/custom/custom-trigger

Reply:
https://github.com/edgexfoundry/edgex-examples/tree/main/application-services/custom/custom-trigger